aws_lambda_builders/workflow.py

```diff
@@ -124,6 +140,12 @@ def __new__(mcs, name, bases, class_dict):
         if not isinstance(cls.CAPABILITY, Capability):
             raise ValueError("Workflow '{}' must register valid capabilities".format(cls.NAME))
 
+        # All workflows must define default value and supported values for build in source
+        if cls.BUILD_IN_SOURCE_BY_DEFAULT is None or cls.BUILD_IN_SOURCE_SUPPORT is None:
```

Comment:
When this case should happen? I thought we will always fallback to default behaviour if something is wrong with the parameters (for also keeping backward compatibility)

Reply:
This is to ensure each workflow class implementation defines these two class constants/attributes. It makes sure that someone doesn't add a new workflow without specifying whether it supports building in source and whether it build in source by default or not.

Reply:
Got it, then can we add unit tests to catch these errors when we introduce a new build workflow? That test can basically try to initialize all workflows and confirm none of them throws this exception.

Reply:
Ah forgot to include a screenshot, but tests already catch this (this logic is called when you initialize the class, I believe):
<img width="818" alt="Screen Shot 2022-12-21 at 10 45 21 AM" src="https://user-images.githubusercontent.com/86501267/208982830-3243784c-cdd5-4a07-81e0-1c02e99d40d7.png">

Reply:
ah I just realized you meant a test that initializes all workflows. 🤔  not sure if it's strictly necessary, as long as the workflow is part of __any__ test (even manual testing), this should be caught.

Reply:
I was referring to a unit test, which will find all workflow definitions and try to initialize them. Any workflow without these values will fail (I wonder if we can enforce this by typing) 🤔 

Can you confirm this part will be validated with integration testing, if so then it shouldn't be a problem for the future?

Reply:
I vote we need the test, we don't explicit check for the ValueError being raised.

Reply:
Since I believe this validation logic runs when the classes are loaded, I think this will be caught by any test. I tested it by adding a new workflow (`MyWorkflow`) without defining the build-in-source-related constants, and I didn't add any new tests. Then I ran `make pr` and it was caught by the first unit test found:
<img width="1557" alt="Screen Shot 2023-01-03 at 1 31 51 PM" src="https://user-images.githubusercontent.com/86501267/210445076-23fe7911-a917-4bb5-8fc5-47fe9a9a23eb.png">

Reply:
Overall, I don't think a test for this is necessary. Otherwise, not really sure what this test would look like. As I can just think of including `import aws_lambda_builders.workflows` in some test and not asserting anything.